middleware.go

```diff
@@ -62,11 +67,31 @@ func (traceConfig *TraceConfig) TraceMiddleware(delegate http.Handler) http.Hand
 func EchoFirstTraceNodeInfo(propagator propagation.TextMapPropagator) func(http.Handler) http.Handler {
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			r, _ = wrphttp.DecodeRequest(r, nil)
+
+			var traceHeaders []string
 			ctx := propagator.Extract(r.Context(), propagation.HeaderCarrier(r.Header))
+			if msg, ok := wrpcontext.Get[*wrp.Message](ctx); ok {
+				traceHeaders = msg.Headers
+			}
+
+			// Iterate through the trace headers (if any), format them, and add them to ctx
+			var tmp propagation.TextMapCarrier = propagation.MapCarrier{}
+			for _, f := range traceHeaders {
+				if f != "" {
+					parts := strings.Split(f, ":")
+					// Remove leading space if there's any
+					parts[1] = strings.Trim(parts[1], " ")
+					tmp.Set(parts[0], parts[1])
+				}
+				ctx = propagation.TraceContext{}.Extract(ctx, tmp)
```

Comment:
Don't you want to do this step , `ctx = propagation.TraceContext{}.Extract(ctx, tmp) `, after going through all the trace headers `f`'s ?

Reply:
Great point. I can definitely just do this once. Updated in latest commit.